File: Code/Python/LAB12_09.py
# ฟังก์ชันบันทึกข้อมูลเริ่มต้นลงไฟล์
def initialize_file():
    with open("data.txt", "w") as file:
        file.write("Apple\nBanana\nCherry\n")

# ฟังก์ชันแสดงข้อมูลจากไฟล์
def display_file():
    print("\n[ข้อมูลปัจจุบันในไฟล์ data.txt]")
    with open("data.txt", "r") as file:
        print(file.read().strip())

# ฟังก์ชันแก้ไขข้อมูลในไฟล์
def replace_word(old_word, new_word):
    with open("data.txt", "r") as file:
        lines = file.readlines()

    with open("data.txt", "w") as file:
        for line in lines:
            file.write(line.replace(old_word, new_word))

    print("[แก้ไขข้อมูลในไฟล์เรียบร้อย!]")

# เรียกใช้งานฟังก์ชัน
initialize_file()  # บันทึกข้อมูลเบื้องต้น
display_file()     # แสดงข้อมูลก่อนแก้ไข

old_word = input("\nEnter word to replace: ")
new_word = input("Enter new word: ")

replace_word(old_word, new_word)
display_file()  # แสดงข้อมูลหลังแก้ไข

File: Code/Cplus/LAB12_09/LAB12_09.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>
using namespace std;

// ฟังก์ชันบันทึกข้อมูลเริ่มต้นลงไฟล์
void initializeFile() {
    ofstream outFile("data.txt");
    if (outFile.is_open()) {
        outFile << "Apple\nBanana\nCherry\n";
        outFile.close();
    }
    else {
        cout << "Error: Cannot create file!" << endl;
    }
}

// ฟังก์ชันแสดงข้อมูลจากไฟล์
void displayFile() {
    ifstream inFile("data.txt");
    if (!inFile.is_open()) {
        cout << "Error: Cannot open file!" << endl;
        return;
    }

    cout << "\n[ข้อมูลปัจจุบันในไฟล์ data.txt]\n";
    string line;
    while (getline(inFile, line)) {
        cout << line << endl;
    }
    inFile.close();
}

// ฟังก์ชันแก้ไขข้อมูลในไฟล์
void replaceWord(const string& oldWord, const string& newWord) {
    ifstream inFile("data.txt");
    if (!inFile.is_open()) {
        cout << "Error: Cannot open file!" << endl;
        return;
    }

    vector<string> lines;
    string line;
    while (getline(inFile, line)) {
        if (line == oldWord) {
            lines.push_back(newWord);
        }
        else {
            lines.push_back(line);
        }
    }
    inFile.close();

    ofstream outFile("data.txt");
    for (const string& updatedLine : lines) {
        outFile << updatedLine << endl;
    }
    outFile.close();
    cout << "[แก้ไขข้อมูลในไฟล์เรียบร้อย!]\n";
}

int main() {
    initializeFile();  // บันทึกข้อมูลเบื้องต้น
    displayFile();     // แสดงข้อมูลก่อนแก้ไข

    string oldWord, newWord;
    cout << "\nEnter word to replace: ";
    cin >> oldWord;
    cout << "Enter new word: ";
    cin >> newWord;

    replaceWord(oldWord, newWord);
    displayFile();  // แสดงข้อมูลหลังแก้ไข

    return 0;
}
